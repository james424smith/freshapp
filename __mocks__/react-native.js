import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
