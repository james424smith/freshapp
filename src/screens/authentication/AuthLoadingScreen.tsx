import React, { useEffect, useCallback } from "react";
import { View, StyleSheet, Platform, Alert } from "react-native";
import { Analytics, Auth } from "aws-amplify";
import {
  FORCE_UPDATE_PASSWORD_ROUTE,
  MAIN_APP_SCREEN_ROUTE,
  SIGN_IN_ROUTE,
  PASSCODE_ROUTE,
  SIGN_UP_ROUTE,
} from "../../constants/routes";
import { getNotificationsDetails } from "../../redux/actions";
import { useDispatch, useSelector } from "react-redux";
import _ from "lodash";
import AsyncStorage from "@react-native-community/async-storage";
import { logoHeightSize, logoWidthSize } from "../../styles/AuthStyles";
import LoadingScreen from "../../components/LoadingScreen";
import {
  StackActions,
  useTheme,
  useNavigation,
} from "@react-navigation/native";
import { IOS } from "../../constants/Platforms";
import { IRootReducerType } from "../../redux/reducers";
import { SeafarerDetails } from "../../interfaces/";
import FingerprintScanner from "react-native-fingerprint-scanner";
import { Appearance } from "react-native-appearance";

const NEW_PASSWORD_CHALLENGE = "NEW_PASSWORD_REQUIRED";

const AuthLoadingScreen = () => {
  const theme = useTheme();
  const inFileStyles = StyleSheet.create({
    root: { flex: 1, backgroundColor: theme.colors.background },
  });
  const navigation = useNavigation();
  const dispatch = useDispatch();

  const seafarer = useSelector<IRootReducerType, SeafarerDetails | undefined>(
    ({ seafarerDetails }) => seafarerDetails.seafarerDetails
  );

  const networkIsConnected = useSelector<IRootReducerType, boolean>(
    ({ network }) => network.isConnected ?? true
  );

  async function getSchemeMode() {
    const getSchemeModeFromStore =
      (await AsyncStorage.getItem("colorScheme")) || "";

    if (!getSchemeModeFromStore || getSchemeModeFromStore === "system") {
      await AsyncStorage.setItem("colorScheme", "system");
      Appearance.set({ colorScheme: Appearance.getColorScheme() });
    } else {
      Appearance.set({ colorScheme: JSON.parse(getSchemeModeFromStore) });
    }
  }

  const findAppropriatePath = useCallback(
    async (userToken: any): Promise<{ stack: string; screen: string }> => {
      const currentUser = await Auth.currentUserInfo();
      const path = await AsyncStorage.getItem("SIGN_IN_PATH");
      const screen = path === "SIGN_IN" ? SIGN_IN_ROUTE : SIGN_UP_ROUTE;
      getSchemeMode();
      if (userToken) {
        if (userToken.challengeName === NEW_PASSWORD_CHALLENGE) {
          return { stack: "Auth", screen: FORCE_UPDATE_PASSWORD_ROUTE };
        }
        if (_.isEmpty(currentUser)) {
          return { stack: "Auth", screen };
        }
        // initial call to get the unread count for the icon badge
        dispatch(getNotificationsDetails());
        /**
         * updates the endpoint in order to associate a pinpoint endpoint with the username
         * instead of the identityId
         */
        await Analytics.updateEndpoint({
          userId: currentUser.username,
          address: await AsyncStorage.getItem("deviceToken"),
          ChannelType: Platform.OS === IOS ? "APNS" : "GCM",
        }).catch((e) => {
          console.log("error", e);
        });

        const biometric = await returnBiometricValue();
        const passcode = await returnPassCode();
        if (biometric) {
          const isBiometricSuccessful = await asyncBiometricLogin();
          if (isBiometricSuccessful) {
            return { stack: "App", screen: MAIN_APP_SCREEN_ROUTE };
          }
          return { stack: "Auth", screen: PASSCODE_ROUTE };
        } else if (passcode) {
          return { stack: "Auth", screen: PASSCODE_ROUTE };
        }
        return { stack: "Auth", screen: PASSCODE_ROUTE };
      } else if (!_.isEmpty(currentUser)) {
        return { stack: "App", screen: MAIN_APP_SCREEN_ROUTE };
      }
      return { stack: "Auth", screen };
    },
    [dispatch]
  );

  async function asyncBiometricLogin() {
    //TODO change let to const
    let isAuthSuccessful = false;

    await FingerprintScanner.isSensorAvailable()
      .then(async (type) => {
        await FingerprintScanner.authenticate({
          description: `Please Log In with ${type}`,
        })
          .then(() => {
            isAuthSuccessful = true;
          })
          .catch(async (error) => {
            const customAlert = () =>
              Alert.alert(
                "Authentication was cancelled",
                "Please try again",
                [
                  {
                    text: "OK",
                    style: "default",
                  },
                ],
                { cancelable: false }
              );
            if (error) {
              customAlert();
              isAuthSuccessful = false;
            }
          });
      })
      .catch((error) => {
        console.log("error", error);
      });

    return isAuthSuccessful;
  }

  // Fetch the token from storage then navigate to our appropriate place
  const _bootstrapAsync = useCallback(async () => {
    const user = await Auth.currentAuthenticatedUser().catch(async () => {
      const path = await AsyncStorage.getItem("SIGN_IN_PATH");
      const authScreen = path === "SIGN_IN" ? SIGN_IN_ROUTE : SIGN_UP_ROUTE;

      navigation.dispatch(StackActions.replace("Auth", { screen: authScreen }));
    });
    const { stack, screen } = await findAppropriatePath(user);

    // This will switch to the App screen or Auth screen and this loading
    // screen will be unmounted and thrown away.
    navigation.dispatch(StackActions.replace(stack, { screen }));
  }, [navigation, findAppropriatePath]);

  const returnBiometricValue = async () => {
    const isBiometricEnabled =
      (await AsyncStorage.getItem("isBiometricEnabled")) || "";

    return isBiometricEnabled ? JSON.parse(isBiometricEnabled) : false;
  };

  const returnPassCode = async () => {
    const getUserPasscodeEnable =
      (await AsyncStorage.getItem("isUserSetPasscode")) || "";

    return getUserPasscodeEnable ? JSON.parse(getUserPasscodeEnable) : false;
  };

  const loginWithoutInternet = useCallback(async () => {
    const biometric = await returnBiometricValue();
    const passcode = await returnPassCode();
    if (biometric) {
      const isBiometricSuccessful = await asyncBiometricLogin();
      if (isBiometricSuccessful) {
        navigation.dispatch(
          StackActions.replace("App", { screen: MAIN_APP_SCREEN_ROUTE })
        );
      }
    } else if (passcode) {
      navigation.dispatch(
        StackActions.replace("Auth", { screen: PASSCODE_ROUTE })
      );
    } else {
      const path = await AsyncStorage.getItem("SIGN_IN_PATH");
      if (path === "SIGN_IN") {
        navigation.dispatch(
          StackActions.replace("Auth", { screen: SIGN_IN_ROUTE })
        );
      } else {
        navigation.dispatch(
          StackActions.replace("Auth", { screen: SIGN_UP_ROUTE })
        );
      }
    }
  }, [navigation]);

  useEffect(() => {
    if (!_.isEmpty(seafarer) && !networkIsConnected) {
      loginWithoutInternet();
    } else {
      _bootstrapAsync().catch((error: any) => console.log(error));
    }
    return () => {
      FingerprintScanner.release();
    };
  }, [
    _bootstrapAsync,
    navigation,
    networkIsConnected,
    seafarer,
    loginWithoutInternet,
  ]);

  // Render any loading content that you like here

  return (
    <View style={inFileStyles.root}>
      <LoadingScreen
        logoHeightSize={logoHeightSize}
        logoWidthSize={logoWidthSize}
      />
    </View>
  );
};

export default AuthLoadingScreen;
