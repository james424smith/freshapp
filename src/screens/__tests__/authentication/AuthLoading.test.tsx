import React from "react";
import "react-native";
import "@testing-library/jest-native/extend-expect";
import AsyncStorage from "@react-native-community/async-storage";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { render, cleanup, waitFor } from "@testing-library/react-native";
import AuthLoadingScreen from "../../authentication/AuthLoadingScreen";
import Amplify, { Analytics, Auth } from "aws-amplify";
import { StackActions } from "@react-navigation/native";
import {
  FORCE_UPDATE_PASSWORD_ROUTE,
  MAIN_APP_SCREEN_ROUTE,
  PASSCODE_ROUTE,
  SIGN_IN_ROUTE,
  SIGN_UP_ROUTE,
} from "../../../constants/routes";
import * as RN from "react-native";
import FingerprintScanner from "react-native-fingerprint-scanner";

Analytics.updateEndpoint = jest.fn().mockResolvedValue({});
Amplify.configure({
  aws_project_region: "eu-west-1",
  aws_cognito_identity_pool_id:
    "eu-west-1:0700dd4d-d3f1-47e8-aeca-c03a916aba97",
  aws_cognito_region: "eu-west-1",
  aws_user_pools_id: "eu-west-1_VdCn6sWLK",
  aws_user_pools_web_client_id: "3uc4kul2beqktthivc578jutfm",
  oauth: {},
  aws_mobile_analytics_app_id: "33c953d8b8a04972904e65148ea64665",
  aws_mobile_analytics_app_region: "eu-west-1",
});
describe("AuthLoadingScreen test", () => {
  beforeEach(() => {
    AsyncStorage.clear();
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render correctly as per the ui made ", () => {
    const mockedState = {
      seafarerDetails: {
        type: "SET_IS_LOADING_DATA",
        payload: { loader: false },
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();
    const MockComponent = () => (
      <Provider store={mockedStore(mockedState)}>
        <AuthLoadingScreen />
      </Provider>
    );

    const { toJSON } = render(<MockComponent />);
    expect(toJSON()).toMatchSnapshot();
  });
  it("should render correctly as per the ui made and network.isConnected is undefined", () => {
    const mockedState = {
      seafarerDetails: {
        type: "SET_IS_LOADING_DATA",
        payload: { loader: false },
      },
      network: { isConnected: undefined },
    };
    const mockedStore = configureMockStore();
    const MockComponent = () => (
      <Provider store={mockedStore(mockedState)}>
        <AuthLoadingScreen />
      </Provider>
    );

    const { toJSON } = render(<MockComponent />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("If biometrics is enabled then we must have value stored in AsyncStorage Corresponding to That.And user will navigate to MAIN_APP_SCREEN_ROUTE ", async () => {
    AsyncStorage.setItem("isBiometricEnabled", "true");
    const result = await AsyncStorage.getItem("isBiometricEnabled");
    expect(result?.toString()).toBe("true");
  });

  it("If PassCode is enabled then we must have value stored in AsyncStorage Corresponding to That.And user will navigate to PASSCODE_ROUTE ", async () => {
    AsyncStorage.setItem("isUserSetPasscode", "true");
    const result = await AsyncStorage.getItem("isUserSetPasscode");
    expect(result?.toString()).toBe("true");
  });

  it("If Neither PassCode nor biometrics is enabled then we must not have value stored in AsyncStorage Corresponding to That. And user will navigate to SIGN_IN_ROUTE ", async () => {
    AsyncStorage.setItem("isUserSetPasscode", "false");
    AsyncStorage.setItem("isBiometricEnabled", "false");
    const result1 = await AsyncStorage.getItem("isUserSetPasscode");
    const result2 = await AsyncStorage.getItem("isBiometricEnabled");

    expect(result1?.toString()).toBe("false");
    expect(result2?.toString()).toBe("false");
  });
});

describe("test findAppropriatePath", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should call the appropriate navigation when userToken.challengeName is NEW_PASSWORD_CHALLENGE", async () => {
    Auth.currentAuthenticatedUser = jest.fn().mockResolvedValue({
      challengeName: "NEW_PASSWORD_REQUIRED",
    });

    const mockedState = {
      seafarerDetails: {
        type: "SET_IS_LOADING_DATA",
        payload: { loader: false },
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();
    const MockComponent = () => (
      <Provider store={mockedStore(mockedState)}>
        <AuthLoadingScreen />
      </Provider>
    );

    const { toJSON } = await waitFor(() => render(<MockComponent />));
    expect(StackActions.replace).toBeCalledWith("Auth", {
      screen: FORCE_UPDATE_PASSWORD_ROUTE,
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it("should call the appropriate navigation when currentUserInfo returns empty object and user opens the app for the first time", async () => {
    Auth.currentAuthenticatedUser = jest
      .fn()
      .mockResolvedValue({ userName: "1234" });
    Auth.currentUserInfo = jest.fn().mockResolvedValue({});

    const mockedState = {
      seafarerDetails: {
        type: "SET_IS_LOADING_DATA",
        payload: { loader: false },
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();
    const MockComponent = () => (
      <Provider store={mockedStore(mockedState)}>
        <AuthLoadingScreen />
      </Provider>
    );

    const { toJSON } = await waitFor(() => render(<MockComponent />));
    expect(StackActions.replace).toBeCalledWith("Auth", {
      screen: SIGN_UP_ROUTE,
    });
    expect(toJSON()).toMatchSnapshot();
  });
  it("should call the appropriate navigation when currentUserInfo returns empty object and user opens the app after a successful signup", async () => {
    Auth.currentAuthenticatedUser = jest
      .fn()
      .mockResolvedValue({ userName: "1234" });
    Auth.currentUserInfo = jest.fn().mockResolvedValue({});
    AsyncStorage.getItem = jest.fn().mockImplementation(async (x) => {
      if (x === "SIGN_IN_PATH") {
        return "SIGN_IN";
      }
    });

    const mockedState = {
      seafarerDetails: {
        type: "SET_IS_LOADING_DATA",
        payload: { loader: false },
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();
    const MockComponent = () => (
      <Provider store={mockedStore(mockedState)}>
        <AuthLoadingScreen />
      </Provider>
    );

    const { toJSON } = await waitFor(() => render(<MockComponent />));
    expect(StackActions.replace).toBeCalledWith("Auth", {
      screen: SIGN_IN_ROUTE,
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it("should call the appropriate navigation when platform is android and updateEndpoint fails.", async () => {
    RN.Platform.OS = "android";
    Auth.currentUserInfo = jest.fn().mockResolvedValue({ userName: "1234" });
    Auth.currentAuthenticatedUser = jest
      .fn()
      .mockResolvedValue({ userName: "1234" });
    Analytics.updateEndpoint = jest.fn().mockRejectedValue({});

    const mockedState = {
      seafarerDetails: {
        type: "SET_IS_LOADING_DATA",
        payload: { loader: false },
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();
    const MockComponent = () => (
      <Provider store={mockedStore(mockedState)}>
        <AuthLoadingScreen />
      </Provider>
    );

    const { toJSON } = await waitFor(() => render(<MockComponent />));
    expect(StackActions.replace).toBeCalledWith("Auth", {
      screen: PASSCODE_ROUTE,
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it("should call the appropriate navigation when biometric is defined but asyncBiometricLogin fails", async () => {
    Auth.currentUserInfo = jest.fn().mockResolvedValue({ userName: "1234" });
    Auth.currentAuthenticatedUser = jest
      .fn()
      .mockResolvedValue({ userName: "1234" });

    AsyncStorage.getItem = jest
      .fn()
      .mockImplementation(async (x) =>
        x === "isUserSetPasscode"
          ? undefined
          : JSON.stringify({ some: "touchId" })
      );
    FingerprintScanner.isSensorAvailable = jest
      .fn()
      .mockResolvedValue("touchId");
    FingerprintScanner.authenticate = jest.fn().mockRejectedValue({});
    const mockedState = {
      seafarerDetails: {
        type: "SET_IS_LOADING_DATA",
        payload: { loader: false },
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();
    const MockComponent = () => (
      <Provider store={mockedStore(mockedState)}>
        <AuthLoadingScreen />
      </Provider>
    );

    const { toJSON } = await waitFor(() => render(<MockComponent />));
    expect(StackActions.replace).toBeCalledWith("Auth", {
      screen: PASSCODE_ROUTE,
    });
    expect(toJSON()).toMatchSnapshot();
  });
  it("should call the appropriate navigation when biometric is defined but isSensorAvailable fails", async () => {
    Auth.currentUserInfo = jest.fn().mockResolvedValue({ userName: "1234" });
    Auth.currentAuthenticatedUser = jest
      .fn()
      .mockResolvedValue({ userName: "1234" });

    AsyncStorage.getItem = jest
      .fn()
      .mockResolvedValue(JSON.stringify({ some: "touchId" }));
    FingerprintScanner.isSensorAvailable = jest.fn().mockRejectedValue({});
    const mockedState = {
      seafarerDetails: {
        type: "SET_IS_LOADING_DATA",
        payload: { loader: false },
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();
    const MockComponent = () => (
      <Provider store={mockedStore(mockedState)}>
        <AuthLoadingScreen />
      </Provider>
    );

    const { toJSON } = await waitFor(() => render(<MockComponent />));
    expect(StackActions.replace).toBeCalledWith("Auth", {
      screen: PASSCODE_ROUTE,
    });
    expect(toJSON()).toMatchSnapshot();
  });
  it("should call the appropriate navigation when isUserSetPasscode is true", async () => {
    Auth.currentUserInfo = jest.fn().mockResolvedValue({ userName: "1234" });
    Auth.currentAuthenticatedUser = jest
      .fn()
      .mockResolvedValue({ userName: "1234" });

    FingerprintScanner.isSensorAvailable = jest.fn().mockRejectedValue({});

    AsyncStorage.getItem = jest
      .fn()
      .mockImplementation(async (x) =>
        x === "isBiometricEnabled" ? "" : "true"
      );

    const mockedState = {
      seafarerDetails: {
        seafarerDetails: { user: "1234" },
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();
    const MockComponent = () => (
      <Provider store={mockedStore(mockedState)}>
        <AuthLoadingScreen />
      </Provider>
    );

    const { toJSON } = await waitFor(() => render(<MockComponent />));
    expect(StackActions.replace).toBeCalledWith("Auth", {
      screen: PASSCODE_ROUTE,
    });
    expect(toJSON()).toMatchSnapshot();
  });
});

describe("test loginWithoutInternet", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should call the appropriate navigation when biometric is available", async () => {
    const mockedState = {
      seafarerDetails: {
        seafarerDetails: { user: "1234" },
      },
      network: { isConnected: false },
    };

    AsyncStorage.getItem = jest
      .fn()
      .mockResolvedValue(JSON.stringify({ some: "touchId" }));

    FingerprintScanner.isSensorAvailable = jest
      .fn()
      .mockResolvedValue("touchId");
    FingerprintScanner.authenticate = jest.fn().mockResolvedValue("touchId");

    const mockedStore = configureMockStore();
    const MockComponent = () => (
      <Provider store={mockedStore(mockedState)}>
        <AuthLoadingScreen />
      </Provider>
    );

    const { toJSON } = await waitFor(() => render(<MockComponent />));
    expect(StackActions.replace).toBeCalledWith("App", {
      screen: MAIN_APP_SCREEN_ROUTE,
    });
    expect(toJSON()).toMatchSnapshot();
  });
  it("should call the appropriate navigation when passcode is available", async () => {
    const mockedState = {
      seafarerDetails: {
        seafarerDetails: { user: "1234" },
      },
      network: { isConnected: false },
    };
    AsyncStorage.getItem = jest
      .fn()
      .mockImplementation(async (x) =>
        x === "isBiometricEnabled" ? "" : "true"
      );

    const mockedStore = configureMockStore();
    const MockComponent = () => (
      <Provider store={mockedStore(mockedState)}>
        <AuthLoadingScreen />
      </Provider>
    );

    const { toJSON } = await waitFor(() => render(<MockComponent />));
    expect(StackActions.replace).toBeCalledWith("Auth", {
      screen: PASSCODE_ROUTE,
    });
    expect(toJSON()).toMatchSnapshot();
  });
  it("should call the appropriate navigation when both biometrics and passcode are undefined and user opens the app for the first time", async () => {
    const mockedState = {
      seafarerDetails: {
        seafarerDetails: { user: "1234" },
      },
      network: { isConnected: false },
    };

    AsyncStorage.getItem = jest.fn().mockResolvedValue("");

    const mockedStore = configureMockStore();
    const MockComponent = () => (
      <Provider store={mockedStore(mockedState)}>
        <AuthLoadingScreen />
      </Provider>
    );

    const { toJSON } = await waitFor(() => render(<MockComponent />));
    expect(StackActions.replace).toBeCalledWith("Auth", {
      screen: SIGN_UP_ROUTE,
    });
    expect(toJSON()).toMatchSnapshot();
  });
  it("should call the appropriate navigation when both biometrics and passcode are undefined after a successful sign up process", async () => {
    const mockedState = {
      seafarerDetails: {
        seafarerDetails: { user: "1234" },
      },
      network: { isConnected: false },
    };
    AsyncStorage.getItem = jest.fn().mockImplementation(async (x) => {
      if (x === "SIGN_IN_PATH") {
        return "SIGN_IN";
      } else {
        return "";
      }
    });

    const mockedStore = configureMockStore();
    const MockComponent = () => (
      <Provider store={mockedStore(mockedState)}>
        <AuthLoadingScreen />
      </Provider>
    );

    const { toJSON } = await waitFor(() => render(<MockComponent />));
    expect(StackActions.replace).toBeCalledWith("Auth", {
      screen: SIGN_IN_ROUTE,
    });
    expect(toJSON()).toMatchSnapshot();
  });
});
