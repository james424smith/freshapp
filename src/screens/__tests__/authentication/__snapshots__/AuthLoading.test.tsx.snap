// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLoadingScreen test should render correctly as per the ui made  1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</View>
`;

exports[`AuthLoadingScreen test should render correctly as per the ui made and network.isConnected is undefined 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</View>
`;

exports[`test findAppropriatePath should call the appropriate navigation when biometric is defined but asyncBiometricLogin fails 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</View>
`;

exports[`test findAppropriatePath should call the appropriate navigation when biometric is defined but isSensorAvailable fails 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</View>
`;

exports[`test findAppropriatePath should call the appropriate navigation when currentUserInfo returns empty object and user opens the app after a successful signup 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</View>
`;

exports[`test findAppropriatePath should call the appropriate navigation when currentUserInfo returns empty object and user opens the app for the first time 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</View>
`;

exports[`test findAppropriatePath should call the appropriate navigation when isUserSetPasscode is true 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</View>
`;

exports[`test findAppropriatePath should call the appropriate navigation when platform is android and updateEndpoint fails. 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</View>
`;

exports[`test findAppropriatePath should call the appropriate navigation when userToken.challengeName is NEW_PASSWORD_CHALLENGE 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</View>
`;

exports[`test loginWithoutInternet should call the appropriate navigation when biometric is available 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</View>
`;

exports[`test loginWithoutInternet should call the appropriate navigation when both biometrics and passcode are undefined after a successful sign up process 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</View>
`;

exports[`test loginWithoutInternet should call the appropriate navigation when both biometrics and passcode are undefined and user opens the app for the first time 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</View>
`;

exports[`test loginWithoutInternet should call the appropriate navigation when passcode is available 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</View>
`;
